tests/test_validation_pandapipes.py

```diff
@@ -0,0 +1,268 @@
+        esdlparser.add_additional_joint()
+
+        # ------------------------------------------------------------------------------
+        # ------------------------------------------------------------------------------
+        # Setup scenario
+        total_producers = len(esdlparser.esdl_asset["heat"]["producer"])
+        total_consumers = len(esdlparser.esdl_asset["heat"]["consumer"])
+
+        # Create panda_pipes network
+        net, net_asset, supply_temperature, return_temperature = esdlparser.createpandapipenet()
+
+        # Setup profile data
+        raw_profile_demand_load_watt = pd.read_csv(
+            os.path.join(base_folder, "input", demand_time_series_file)
+        )
+        profile_demand_load_watt = pd.DataFrame(columns=["0"])
+        profile_demand_load_watt["0"] = raw_profile_demand_load_watt["demand_1"]
+
+        demand_power = profile_demand_load_watt["0"].to_numpy() / total_consumers
+        profile_demand_load_watt = pd.DataFrame(results["demand_1.Heat_demand"])
+
+        # Setup supply mass flow for panda_pipes
+        average_temperature_kelvin = (supply_temperature + return_temperature) / 2.0 + 273.15
+        cp_joule_kgkelvin = pp.get_fluid(net).get_heat_capacity(average_temperature_kelvin)
+
+        # Enforce mass flow rate instead of cacluting it from Q = m_dot...
+        mesido_demand_flow_kg_s = results["Pipe1.Q"] * 988.0
+        demand_flow = mesido_demand_flow_kg_s
+        supply_flow = demand_flow * total_consumers / total_producers
+
+        # Assign profiles to assets
+        net.flow_control.control_active[0] = False  # 1st supplier
+        net.flow_control.control_active[0 + total_producers] = False  # 1st demand
+
+        # Producer
+        isupply = 0
+        supply_flow_kg_s = pd.DataFrame(supply_flow, columns=[f"{isupply}"])
+        ds_supply_pump_flow_kg_s = DFData(supply_flow_kg_s)
+        control.ConstControl(
+            net,
+            element="circ_pump_mass",
+            variable="mdot_flow_kg_per_s",
+            element_index=net.circ_pump_mass.index.values[isupply],
+            data_source=ds_supply_pump_flow_kg_s,
+            profile_name=net.circ_pump_mass.index.values[isupply].astype(str),
+        )
+        supply_temp_kelvin = pd.DataFrame(
+            [supply_temperature + 273.15] * len(profile_demand_load_watt),
+            columns=[f"{isupply}"],
+        )
+        df_supply_pump_temperature_kelvin = DFData(supply_temp_kelvin)
+        control.ConstControl(
+            net,
+            element="circ_pump_mass",
+            variable="t_flow_k",
+            element_index=net.circ_pump_mass.index.values[isupply],
+            data_source=df_supply_pump_temperature_kelvin,
+            profile_name=net.circ_pump_mass.index.values[isupply].astype(str),
+        )
+        supply_flow_kgs_s = pd.DataFrame(supply_flow, columns=[f"{isupply}"])
+        ds_supply_control_flow_kgs_s = DFData(supply_flow_kgs_s)
+        control.ConstControl(
```

Comment:
for me it is unclear why this additional ConstControl is needed? Is this not already set by the first ConstControl

Reply:
Deleted. More info if you are interested: The control.ConstControl with element="flow_control" is not needed for this simple case. It is however needed for when there is more than 1 producer/heat demand. So I deleted the code for this test. I used this in my other runs when I compared chess with pandapipes for ring networks with 2 heat demands and 2 producers. Then you need to specify these controls and also deactivate it for two assets (1 heat demand and 1 producer) to ensure the system is not over constraints. If I remember correctly.

Reply:
check